text/0000-rust-spec.md

```diff
@@ -0,0 +1,171 @@
+
+Authors of unsafe code could benefit a lot from clear definitions of what is and isn't undefined behaviour.
+Safety critical Rust software won't pass certification without a specification that clearly specifies how Rust code behaves.
+Proposals and discussions about new Rust language features could be more efficient and precise
+using accurately defined terms that everyone agrees on.
+Questions about subtle interactions between features of the language could be answered
+using precise information from a specification, instead a combination of guesses and several non-authoritative sources.
+
+# Current state
+
+Languages like C and C++ are standardized.
+Rust is not. Standardization comes down to, basically:
+
+1. Having an accurate specification (a document)
+2. An (open) process for evolution of the language
+3. Stability
+
+Rust currently already has 2 and 3, but not 1.
+
+For 1, we currently have:
+the (incomplete) [Rust Reference](https://doc.rust-lang.org/stable/reference/),
+the [Standard Library Reference Documentation](https://doc.rust-lang.org/stable/std/),
+the [Rust Nomicon](https://doc.rust-lang.org/nightly/nomicon/),
+the [Unsafe Code Guidelines Project](https://github.com/rust-lang/unsafe-code-guidelines/),
+[Miri](https://github.com/rust-lang/miri/),
+the collection of [accepted RFCs](https://rust-lang.github.io/rfcs/),
+the [Ferrocene Language Specification](https://spec.ferrocene.dev/),
+lots context and decisions spread over [tons of GitHub issues](https://github.com/rust-lang/rust/issues/),
+[MiniRust](https://github.com/RalfJung/minirust),
+the [source code](https://github.com/rust-lang/rust/),
+and more.
+
+These are currently all incomplete, and/or not a good source to rely on.
+
+More background information is available in [this blog post](https://blog.m-ou.se/rust-standard/).
+
+# Goals and non-goals
+
+- The goal of the Rust specification work is the creation of a document, the Rust specification.
+
+- The goal is _not_ to change how the language evolves;
+  the relevant teams (Language, Libs-API, …) remain in charge of the evolution of their respective parts of Rust,
+  and will continue to use processes as they see fit (e.g. RFCs).
+
+- The specification will only be considered "official" once the relevant teams have approved of its contents.
+  Changes to the official specification must be approved by the relevant team(s).
+
+- The goal is to serve the needs of Rust users, such as
+  authors of unsafe Rust code, those working on safety critical Rust software,
+  language designers, maintainers of Rust tooling, and so on.
+
+- It is _not_ a primary goal of the specification to aid in the development of alternative Rust implementations,
+  although authors of alternative compilers might still find the specification to be useful.
+
+  What this means is that, unlike the C or C++ standard,
+  the Rust specification does not provide a set of requirements for a compiler to be able to call itself "a Rust™ compiler".
```

Comment:
Complete nit, but Rust is a registered trademark, so it should be Rust®, not Rust™.